projects/storefrontlib/src/shared/components/media/media.service.ts

```diff
@@ -1,108 +1,133 @@
-import { Injectable } from '@angular/core';
-import { OccConfig } from '@spartacus/core';
+import { Inject, Injectable } from '@angular/core';
+import { Config, Image, OccConfig } from '@spartacus/core';
 import { BreakpointService } from '../../../layout/breakpoint/breakpoint.service';
-import { BREAKPOINT } from '../../../layout/config/layout-config';
-import { Media, MediaFormats } from './media.model';
-
-/** the default format is used for browsers that do not support   */
-const DEFAULT_MEDIA_FORMAT = 'tablet';
+import { StorefrontConfig } from '../../../storefront-config';
+import { MediaConfig } from './media.config';
+import { Media, MediaContainer, MediaFormatSize } from './media.model';
 
+/**
+ * Service which generates media URLs. It leverage the MediaContainer and MediaFormats so
+ * that URLs and sizes are generated for the same media. This helps to improve performance
+ * across difference devices and layouts.
+ *
+ * Media formats are optional, but highly recommended. The format will help the browser to
+ * identify the right media for the right experience.
+ *
+ * The MediaService will generate absolute URLs in case relative URLs are provided for the Media.
+ * The baseUrl is read from the `occConfig.backend.media.baseUrl` or
+ * `occConfig.backend.occ.baseUrl`.
+ */
 @Injectable({
   providedIn: 'root',
 })
 export class MediaService {
+  /**
+   * The media formats sorted by size. The media format representing the smallest
+   * size is sorted on top.
+   */
+  protected sortedFormats: { code: string; size: MediaFormatSize }[];
+  protected reversedFormats: { code: string; size: MediaFormatSize }[];
+
   constructor(
-    protected config: OccConfig,
+    @Inject(Config) protected config: StorefrontConfig,
+    /**
+     * The BreakpointService is no longer used in version 2.0 as the different size formats are
+     * driven by configuration only. There's however a change that this service will play a role
+     * in the near future, which is why we keep the constructor as-is.
+     */
     protected breakpointService: BreakpointService
-  ) {}
-
-  private get mediaFormats(): MediaFormats[] {
-    return [
-      {
-        code: 'mobile',
-        threshold: this.breakpointService.getSize(BREAKPOINT.xs),
-      },
-      {
-        code: 'tablet',
-        threshold: this.breakpointService.getSize(BREAKPOINT.sm),
-      },
-      {
-        code: 'desktop',
-        threshold: this.breakpointService.getSize(BREAKPOINT.md),
-      },
-      {
-        code: 'widescreen',
-        threshold: this.breakpointService.getSize(BREAKPOINT.lg),
-      },
-    ];
+  ) {
+    this.sortFormats();
```

Comment:
As configuration might not be final at constructor time, we should consider either using `ConfigInitializerService.isStable()` or `getStableConfig`, or making this logic to run at the first time it's actually needed